docs/lang-ref-structs.md

```diff
@@ -0,0 +1,103 @@
+### Structs
+
+Structs are well-defined data objects, comprised of key/value pairs known as "attributes". They are a way to store and refer to data of a known structure.
+
+The most commonly used `struct` is `data.values`, supplied by the [`@ytt:data`](ytt-data-values.md) module.
+
+For example, a data values defined by:
+
+```yaml
+#@data/values
+---
+db_conn:
+  host: acme.example.com
+```
+
+is automatically processed into a `struct` (named `values`): the keys in the `@data/values` file are defined one-for-one as attributes on the `struct`.
+ 
+Those attribues can be referenced by name:
+
+```yaml
+#@ load("@ytt:data", "data")
+---
+persistence:
+  db_url: #@ data.values.db_conn.host
+```
+
+_([ytt @data/values](ytt-data-values.md) describes this process in more detail.)_
+
+__
+
+Described below:
+- [Attributes](#attributes) — how to refer to a datum within a `struct`
+- [Build-in Functions](#built-in-functions) — functions that support using a `struct`
+- [Creating `struct`s](#creating-structs) — support for creating your own `struct`s
+
+---
+
+#### Attributes
+
+Attributes are a key/value pair, where the key is a `string` and the value can be of any type.
+
+Attributes can be referenced:
+- by name (using "[dot notation](https://github.com/google/starlark-go/blob/master/doc/spec.md#dot-expressions)")
+    ```yaml
+    db_url: #@ data.values.db_conn.host
+    ```
+- by string (using "[index notation](https://github.com/google/starlark-go/blob/master/doc/spec.md#index-expressions)") _(as of v0.31.0)_:
```

Comment:
Nit: This is also accessing by name, it's changing the notation not the access method.

Reply:
nice catch. Will use "field" there.